src/Controls/tests/Xaml.UnitTests/Issues/Maui32597.xaml.cs

```diff
@@ -0,0 +1,58 @@
+using System.Collections.ObjectModel;
+using Xunit;
+
+namespace Microsoft.Maui.Controls.Xaml.UnitTests;
+
+public partial class Maui32597 : ContentPage
+{
+	public Maui32597()
+	{
+		InitializeComponent();
+	}
+
+	[Collection("Issue")]
+	public class Tests
+	{
+		[Theory]
+		[XamlInflatorData]
+		internal void BindingToObservablePropertyFromCamelCaseField(XamlInflator inflator)
+		{
+			// Test binding to property generated from camelCase field 'name'
+			var page = new Maui32597(inflator);
+			Assert.Equal("Test Name", page.NameLabel.Text);
+		}
+
+		[Theory]
+		[XamlInflatorData]
+		internal void BindingToObservablePropertyFromUnderscorePrefixedField(XamlInflator inflator)
+		{
+			// Test binding to property generated from _camelCase field '_title'
+			var page = new Maui32597(inflator);
+			Assert.Equal("Test Title", page.TitleLabel.Text);
+		}
+
+		[Theory]
+		[XamlInflatorData]
+		internal void BindingToObservablePropertyCollection(XamlInflator inflator)
+		{
+			// Test binding to collection property generated from ObservableProperty
+			var page = new Maui32597(inflator);
+			Assert.Equal("3", page.TagCountLabel.Text);
+		}
+	}
+}
+
+// Mock ViewModel with ObservableProperty attributes
+// Note: In real usage, these would be source-generated by CommunityToolkit.Mvvm
+// For testing, we manually define the properties that would be generated
+public class Maui32597ViewModel
+{
+	// These would normally be:
+	// [ObservableProperty]
+	// private string name;
+	// But for testing, we provide the generated properties directly
+
+	public string Name => "Test Name";
+	public string Title => "Test Title";
+	public ObservableCollection<string> Tags { get; } = new() { "Tag1", "Tag2", "Tag3" };
+}
```

Comment:
This does not test the new code at all. We would need to have the actual fields with attributes here. To make this fully working would mean to include the Community Tookit source generator or mock the source generator. I don't know if that's a good idea though. We should probably just revert this.